internal/provider/config/config.go

```diff
@@ -1,28 +1,61 @@
 package config
 
-type Kairos struct {
+type P2P struct {
 	NetworkToken string `yaml:"network_token,omitempty"`
 	NetworkID    string `yaml:"network_id,omitempty"`
 	Role         string `yaml:"role,omitempty"`
 	DNS          bool   `yaml:"dns,omitempty"`
 	LogLevel     string `yaml:"loglevel,omitempty"`
-	Hybrid       bool   `yaml:"hybrid,omitempty"`
+	VPN          VPN    `yaml:"vpn,omitempty"`
+
 	MinimumNodes int    `yaml:"minimum_nodes,omitempty"`
+	SkipAuto     bool   `yaml:"skip_auto"`
+	DisableDHT   bool   `yaml:"disable_dht,omitempty"`
+	AutoHA       AutoHA `yaml:"auto-ha,omitempty"`
+
+	DynamicRoles bool `yaml:"dynamic_roles,omitempty"`
+}
+
+type VPN struct {
+	Create *bool             `yaml:"create,omitempty"`
+	Use    *bool             `yaml:"use,omitempty"`
+	Env    map[string]string `yaml:"env,omitempty"`
+}
+
+// If no setting is provided by the user,
+// we assume that we are going to create and use the VPN
+// for the network layer of our cluster.
+func (p P2P) UseVPNWithKubernetes() bool {
+	return p.VPNNeedsCreation() && (p.VPN.Use == nil || *p.VPN.Use)
+}
+
+func (p P2P) VPNNeedsCreation() bool {
+	return p.VPN.Create == nil || *p.VPN.Create
 }
 
 type Config struct {
-	Kairos   *Kairos           `yaml:"kairos,omitempty"`
-	K3sAgent K3s               `yaml:"k3s-agent,omitempty"`
-	K3s      K3s               `yaml:"k3s,omitempty"`
-	VPN      map[string]string `yaml:"vpn,omitempty"`
-	KubeVIP  KubeVIP           `yaml:"kubevip,omitempty"`
+	P2P      *P2P    `yaml:"p2p,omitempty"`
+	K3sAgent K3s     `yaml:"k3s-agent,omitempty"`
+	K3s      K3s     `yaml:"k3s,omitempty"`
+	KubeVIP  KubeVIP `yaml:"kubevip,omitempty"`
 }
 
 type KubeVIP struct {
 	Args        []string `yaml:"args,omitempty"`
 	EIP         string   `yaml:"eip,omitempty"`
 	ManifestURL string   `yaml:"manifest_url,omitempty"`
 	Interface   string   `yaml:"interface,omitempty"`
+	Enable      bool     `yaml:"enable,omitempty"`
+}
+
+func (k KubeVIP) IsEnabled() bool {
+	return k.Enable || k.EIP != ""
```

Comment:
NOTE: If `EIP` is mandatory, then we don't need `Enable`. And if the user sets it to `false` and then set `EIP`, the fact that eventually it's enabled (because of the code here), it makes it more confusing.

As we discussed in the call, let do this:

- Enable become a pointer to boolean and it's `true` by default.
- This method normally should check only the value of `Enable`. This means, if it's explicitly set to `false` then we don't check the `EIP` value. That's because, in the future, the user may be able to set it to empty string and we could implement some auto-discovery of the IP address. But until we do, let' keep the `EIP` in this check here but with a comment explaining that.